bigquery/gcloud/aio/bigquery/table.py

```diff
@@ -0,0 +1,315 @@
+    def _make_insert_body(
+            rows: List[Dict[str, Any]], *, skip_invalid: bool,
+            ignore_unknown: bool, template_suffix: Optional[str],
+            insert_id_fn: Callable[[Dict[str, Any]], str]) -> Dict[str, Any]:
+        body = {
+            'kind': 'bigquery#tableDataInsertAllRequest',
+            'skipInvalidRows': skip_invalid,
+            'ignoreUnknownValues': ignore_unknown,
+            'rows': [{
+                'insertId': insert_id_fn(row),
+                'json': row,
+            } for row in rows],
+        }
+
+        if template_suffix is not None:
+            body['templateSuffix'] = template_suffix
+
+        return body
+
+    def _make_load_body(
+            self, source_uris: List[str], project: str, autodetect: bool,
+            source_format: SourceFormat,
+            write_disposition: Disposition,
+            ignore_unknown_values: bool,
+            schema_update_options: List[SchemaUpdateOption]
+    ) -> Dict[str, Any]:
+        return {
+            'configuration': {
+                'load': {
+                    'autodetect': autodetect,
+                    'ignoreUnknownValues': ignore_unknown_values,
+                    'sourceUris': source_uris,
+                    'sourceFormat': source_format.value,
+                    'writeDisposition': write_disposition.value,
+                    'schemaUpdateOptions': [
+                        e.value for e in schema_update_options],
+                    'destinationTable': {
+                        'projectId': project,
+                        'datasetId': self.dataset_name,
+                        'tableId': self.table_name,
+                    },
+                },
+            },
+        }
+
+    def _make_query_body(
+            self, query: str, project: str,
+            write_disposition: Disposition,
+            use_query_cache: bool,
+            dry_run: bool) -> Dict[str, Any]:
+        return {
+            'configuration': {
+                'query': {
+                    'query': query,
+                    'writeDisposition': write_disposition.value,
+                    'destinationTable': {
+                        'projectId': project,
+                        'datasetId': self.dataset_name,
+                        'tableId': self.table_name,
+                    },
+                    'useQueryCache': use_query_cache,
+                },
+                'dryRun': dry_run,
+            },
+        }
+
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/tables/insert
+    async def create(self, table: Dict[str, Any],
+                     session: Optional[Session] = None,
+                     timeout: int = 60) -> Dict[str, Any]:
+        """Create the table specified by tableId from the dataset."""
+        project = await self.project()
+        url = (f'{API_ROOT}/projects/{project}/datasets/'
+               f'{self.dataset_name}/tables')
+        if 'tableReference' not in table:
```

Comment:
Looks like this would be another place to overwrite `tableReference` unconditionally.

Reply:
done.